### PR TITLE
Add get_file operations for caching testdata files

### LINKED ISSUE #16: Re-enable tests
### Description
- Port https://github.com/Ouranosinc/raven/pull/322 here
- Use new functionality to download test files locally, replacing links in TESTDATA. 
- Confirm all tests pass

### PR DESCRIPTION
**Pull request by Zeitsperre**
Fixes #16

## What's here?

- This ports the `raven` WPS project's `get_file` operations and implements them in the testing and file caching for the testing ensemble of RavenPy

## What else?

- I've pinned Click below 8.0 as this is not yet compatible with `rasterio`, `xarray` and others.

### CODE REVIEW

File: tests/common.py [hunk 1]
```diff
@@ -210,7 +223,21 @@ def _convert_3d(fn):
     ds = xr.open_dataset(fn).rename({"nstations": "lat"})
 
     out = xr.Dataset(
-        coords={"lon": (["lon",], lon), "lat": (["lat",], lat), "time": ds.time}
+        coords={
```
Comment (huard) [inline tests/common.py:226]: remove trailing , to appease black

File: tests/common.py [hunk 2]
```diff
@@ -191,7 +198,13 @@ def _convert_2d(fn):
 
     # Add geometry feature variables
     for key, val in features.items():
-        ds[key] = xr.DataArray(name=key, data=[val,], dims=("region",),)
+        ds[key] = xr.DataArray(
```
Comment (huard) [inline tests/common.py:201]: black overdoes it

File: tests/common.py [hunk 3]
```diff
@@ -21,12 +22,18 @@
...
-TESTDATA["raven-gr4j-cemaneige-nc-ts"] = (
-    TD / "raven-gr4j-cemaneige" / "Salmon-River-Near-Prince-George_meteo_daily.nc"
-)
-TESTDATA["raven-gr4j-cemaneige-nc-rv"] = tuple(
-    (TD / "raven-gr4j-cemaneige").glob("raven-gr4j-salmon.rv?")
-)
+TESTDATA[
+    "raven-gr4j-cemaneige-nc-ts"
+] = "raven-gr4j-cemaneige/Salmon-River-Near-Prince-George_meteo_daily.nc"
+TESTDATA["raven-gr4j-cemaneige-nc-rv"] = query_folder("raven-gr4j-cemaneige", pattern=".rv")
```
Comment (huard) [inline tests/common.py:28]: @cjauvin do you think for the test suite we should replace the TESTDATA dict by pytest fixtures calling get_file and query_folder ?

**Review by huard — APPROVED** [review]

**Comment by Zeitsperre** [comment]
For re-enabling all the tests, I'll open a new PR. I think the tools are there and working as intended. We just need to implement them in the way we'd like. Good to merge?

**Pull request merged**